src/devices/Common/Iot/Device/Multiplexing/IOutputSegment.cs

```diff
@@ -0,0 +1,63 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Device.Gpio;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Iot.Device.Multiplexing
+{
+    /// <summary>
+    /// Abstracts a segment of outputs from multiplexing sources (like a shift register).
+    /// </summary>
+    public interface IOutputSegment : IDisposable
+    {
+        /// <summary>
+        /// Length of segment (number of outputs)
+        /// </summary>
+        int Length { get; }
+
+        /// <summary>
+        /// Segment values.
+        /// </summary>
+        PinValue this[int index] { get; set; }
+
+        /// <summary>
+        /// Writes a PinValue to a virtual segment.
+        /// Does not display output.
+        /// </summary>
+        void Write(int index, PinValue value);
+
+        /// <summary>
+        /// Writes discrete underlying bits to a virtual segment.
+        /// Writes each bit, left to right. Least significant bit will written to index 0.
+        /// Does not display output.
+        /// </summary>
+        void Write(byte value);
+
+        /// <summary>
+        /// Writes discrete underlying bits to a virtual output.
+        /// Writes each byte, left to right. Least significant bit will written to index 0.
+        /// Does not display output.
+        /// </summary>
+        void Write(ReadOnlySpan<byte> value);
```

Comment:
Sorry, now that I saw this perhaps instead of ReadOnlySpan<byte> this should be ReadOnlySpan<PinValue>? The advantage of changing it is that PinValue can be implicitly casted from byte, bool, or int.

Reply:
I enjoy using binary literals because it is such a compact syntax. Also, the important states have such easy to remember (even more terse) representations: `0`, `255`. How about we sit on this one a little longer.  I know we've discussed it before. I'm torn.  It probably makes sense to look at some Python examples to see how representing multiple bits is handled.